### PR TITLE
Support requiring a call context for all calls

### PR DESCRIPTION
**Pull request by KenKozman-GE**
This will be used to support the EMS API usage mode where we want to use trusted and password auth in the same process. By requiring a call context we can help force the caller to consider which style of auth they want to use.

### CODE REVIEW

File: src/Client/V2/Service/MessageHandler.cs [hunk 1]
```diff
@@ -137,6 +137,8 @@ protected override async Task<HttpResponseMessage> SendAsync( HttpRequestMessage
             AddCustomHeaders( request.Headers );
 
             CallContext ctx = RetrieveCallContext( request );
+            if( ctx == null && m_serviceConfig.RequireCallContext )
+                throw new EmsApiNoCallContextException();
```
Comment (c-owens) [inline src/Client/V2/Service/MessageHandler.cs:141]: Very minor but I would prefer a blank line after the no-brace if statement when the following line is unrelated

File: src/Tests/ErrorHandlingTests.cs [hunk 2]
```diff
@@ -35,5 +35,15 @@ public void Disabling_api_exceptions_should_not_throw_an_exception()
...
         }
+
+        [Fact( DisplayName = "Requiring a call context but not providing one should throw" )]
+        public void Require_call_context_but_dont_provide_one()
+        {
+            var config = m_config.Clone();
+            config.RequireCallContext = true;
+            using var service = new EmsApiService( config );
+            Action causeFailure = () => service.EmsSystem.Get();
+            causeFailure.Should().Throw<EmsApiException>();
```
Comment (c-owens) [inline src/Tests/ErrorHandlingTests.cs:46]: Might be better to test your specific exception here?

**Review by c-owens — CHANGES_REQUESTED** [review]

File: src/Client/V2/Service/MessageHandler.cs (hunk 1 above)
Comment (KenKozman-GE) [reply src/Client/V2/Service/MessageHandler.cs:141]: Good stuff; fixing.

File: src/Tests/ErrorHandlingTests.cs (hunk 2 above)
Comment (KenKozman-GE) [reply src/Tests/ErrorHandlingTests.cs:46]: Yeah I had that initially and it did not work, but I was too lazy to dive into it, will do that next.

**Review by c-owens — APPROVED** [review]
👍
